runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/grpcclient/GrpcWindmillServer.java

```diff
@@ -295,10 +294,12 @@ private synchronized CloudWindmillServiceV1Alpha1Grpc.CloudWindmillServiceV1Alph
     if (stubList.isEmpty()) {
       throw new RuntimeException("windmillServiceEndpoint has not been set");
     }
-    if (stubList.size() == 1) {
-      return stubList.get(0);
-    }
-    return stubList.get(rand.nextInt(stubList.size()));
+
+    CloudWindmillServiceV1Alpha1Grpc.CloudWindmillServiceV1Alpha1Stub stub =
+        (stubList.size() == 1 ? stubList.get(0) : stubList.get(rand.nextInt(stubList.size())));
+
+    return stub.withDeadlineAfter(
+        AbstractWindmillStream.DEFAULT_STREAM_RPC_DEADLINE_SECONDS, TimeUnit.SECONDS);
```

Comment:
see other cl about withDeadlineAfter

either use some other mechanism or document that result of stub() should just be used and not cached.

Reply:
could name stubWithAbsoluteDeadlineSet to make it obvious to callers instead of a comment.

Reply:
changed method name, and added ClientInterceptor

Reply:
i rebased this branch onto the branch that removes the sync stubs so i could reuse some of the code.

Reply:
done, decided to keep it the same and change the name for simplicity

Reply:
Looking again, the AbstractWindmillStream are used for longer than the deadline, they reconnect grpc streams internally.  So I think you need to use the observer approach and not a fixed deadline.